src/configuration.ts

```diff
@@ -397,21 +399,34 @@ function getLaunchConfiguration(name: string): LaunchConfiguration | undefined {
 // Read the identifier from workspace state storage, then find the corresponding object
 // in the launch configurations array from settings.
 // Also update the status bar item.
-function readCurrentLaunchConfiguration(): void {
+async function readCurrentLaunchConfiguration(): Promise<void> {
     readLaunchConfigurations();
     let currentLaunchConfigurationName: string | undefined = extension.getState().launchConfiguration;
     if (currentLaunchConfigurationName) {
         currentLaunchConfiguration = getLaunchConfiguration(currentLaunchConfigurationName);
     }
 
+    let launchConfigStr : string = "No launch configuration set.";
     if (currentLaunchConfiguration) {
-        let launchConfigStr: string = launchConfigurationToString(currentLaunchConfiguration);
+        launchConfigStr = launchConfigurationToString(currentLaunchConfiguration);
         logger.message(`Reading current launch configuration "${launchConfigStr}" from the workspace state.`);
         statusBar.setLaunchConfiguration(launchConfigStr);
     } else {
-        logger.message("No current launch configuration is set in the workspace state.");
-        statusBar.setLaunchConfiguration("No launch configuration set.");
+        // A null launch configuration after a non empty launch configuration string name
+        // means that the name stored in the project state does not match any of the entries in settings.
+        // This typically happens after the user modifies manually "makefile.launchConfigurations"
+        // in the .vscode/settings.json, specifically the entry that corresponds to the current launch configuration.
+        // Make sure to unset the launch configuration in this scenario.
+        if (currentLaunchConfigurationName !== undefined && currentLaunchConfigurationName !== "") {
+            logger.message(`Launch configuration "${currentLaunchConfigurationName}" is no longer defined in settings "makefile.launchConfigurations".`);
+            await setLaunchConfigurationByName("");
+        } else {
+            logger.message("No current launch configuration is set in the workspace state.");
+        }
     }
+
+    statusBar.setLaunchConfiguration(launchConfigStr);
```

Comment:
this looks like it is called already on line 413.  Does it need to be called again here?

Reply:
Actually it needs to be called here and not on 413, thanks for noticing this.